app/controllers/child_objects_controller.rb

```diff
@@ -70,7 +70,14 @@ def destroy
 
     # Use callbacks to share common setup or constraints between actions.
     def set_child_object
-      @child_object = ChildObject.find(params[:id]) if params[:id]
+      begin
+        @child_object = ChildObject.find(params[:id]) if params[:id]
+      rescue ActiveRecord::RecordNotFound
+        respond_to do |format|
+          format.html { redirect_to parent_objects_url, notice: "Child object, oid: #{params[:id]}, was not found in local instance." }
+          format.json { head :no_content }
+        end
+      end
```

Comment:
No real reason to raise an exception here. 

```suggestion
      @child_object = ChildObject.find_by(id: params[:id]) if params[:id]
      if @child_object.blank?
        respond_to do |format|
          format.html { redirect_to parent_objects_url, notice: "Child object, oid: #{params[:id]}, was not found in local instance." }
          format.json { head :no_content }
        end
      end
```

Reply:
We attempted to use the `find_by` method but that resulted in the error below.  So we stuck with what we had.  Open to trying again though.

![image](https://user-images.githubusercontent.com/36549923/127715780-8ba793aa-1108-4bfc-bc16-9c6573ea01ed.png)